### PR TITLE
Feature v0.14.9 node disconnect

### PR DESCRIPTION
**Pull request by 0xdcarns**

### CODE REVIEW

**Comment by 0xdcarns** [comment]
Added command to connect or disconnect. run with `netclient <disconnect|connect> -n network`
Test: 
- Setup dev netmaker server
- Connect a client normally
- Run disconnect command, see that no WG interface get's created and ensure it doesn't get created after checkin
- Run connect command, see that WG interface is created and ensure it stays created after checkin

Needs: Add to GUI if it works

File: netclient/wireguard/common.go [hunk 1]
```diff
@@ -194,10 +194,7 @@ func InitWireguard(node *models.Node, privkey string, peers []wgtypes.PeerConfig
 		}
 	}
 	logger.Log(1, "interface ready - netclient.. ENGAGE")
-	if syncconf { // should never be called really.
-		fmt.Println("why here")
-		err = SyncWGQuickConf(ifacename, confPath)
-	}
+
```
Comment (mattkasun) [inline netclient/wireguard/common.go:197]: if syncconf check is removed the corresponding func arg should be removed and all calls to InitWireguard updated.

File: netclient/wireguard/unix.go [hunk 2]
```diff
@@ -24,49 +22,55 @@ func ApplyWGQuickConf(confPath string, ifacename string) error {
...
-		logger.Log(0, "error syncing conf, resetting")
-		err = ApplyWGQuickConf(confPath, iface)
-	}
-	errN := os.Remove(tmpConf)
-	if errN != nil {
-		logger.Log(0, errN.Error())
-	}
-	return err
-}
+// SyncWGQuickConf - formats config file and runs sync command - DEPRECATED
```
Comment (mattkasun) [inline netclient/wireguard/unix.go:45]: delete

**Review by mattkasun — CHANGES_REQUESTED** [review]
with the changes to ApplyXXXConf a slight refactor could eliminate all the RemoveXXXConf functions.  just call ApplyXXXConf with isConnected = false

**Review by afeiszli — APPROVED** [review]

File: netclient/wireguard/common.go [hunk 3]
```diff
@@ -252,9 +249,9 @@ func SetWGConfig(network string, peerupdate bool, peers []wgtypes.PeerConfig) er
 		}
 		err = SetPeers(iface, &cfg.Node, peers)
 	} else if peerupdate {
-		err = InitWireguard(&cfg.Node, privkey, peers, true)
+		err = InitWireguard(&cfg.Node, privkey, peers)
 	} else {
-		err = InitWireguard(&cfg.Node, privkey, peers, false)
```
Comment (mattkasun) [inline netclient/wireguard/common.go:257]: with this change if statement is no longer requiec

**Review by mattkasun — CHANGES_REQUESTED** [review]

**Review by abhishek9686 — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]
This is a pass, but I wanted to note that the journal shows logs that are expected from a disconnected client but a user might think is an error. example: Aug 24 03:14:17 egressQA netclient[3005]: [netclient] 2022-08-24 03:14:17 [localport.go-26] GetLocalListenPort(): failed to parse interface nm-disconnect
Aug 24 03:14:17 egressQA netclient[3005]: [netclient] 2022-08-24 03:14:17 [localport.go-38] UpdateLocalListenPort(): network: disconnect error encountered checking local listen port:  nm-disconnect file does not exist

**Review by mattkasun — APPROVED** [review]

**Comment by afeiszli** [comment]
Should we go ahead and merge this for 0.15.1?

**Pull request merged**